mlir/lib/Target/LLVMIR/ModuleImport.cpp

```diff
@@ -554,13 +554,277 @@ static Attribute convertCGProfileModuleFlagValue(ModuleOp mlirModule,
+    return failure();
+  };
+
+  // Build ModuleFlagProfileSummaryAttr by sequentially fetching elements in
+  // a fixed order: format, total count, etc.
+  SmallVector<Attribute> profileSummary;
+  std::optional<ProfileSummaryFormatKind> format = convertProfileSummaryFormat(
+      mlirModule, llvmModule, mdTuple->getOperand(summayIdx++));
+  if (!format.has_value())
+    return nullptr;
+
+  uint64_t totalCount = 0, maxCount = 0, maxInternalCount = 0,
+           maxFunctionCount = 0, numCounts = 0, numFunctions = 0;
+  if (!convertInt64FromKeyValueTuple(mlirModule, llvmModule,
+                                     mdTuple->getOperand(summayIdx++),
+                                     "TotalCount", totalCount))
+    return nullptr;
+  if (!convertInt64FromKeyValueTuple(mlirModule, llvmModule,
+                                     mdTuple->getOperand(summayIdx++),
+                                     "MaxCount", maxCount))
+    return nullptr;
+  if (!convertInt64FromKeyValueTuple(mlirModule, llvmModule,
+                                     mdTuple->getOperand(summayIdx++),
+                                     "MaxInternalCount", maxInternalCount))
+    return nullptr;
+  if (!convertInt64FromKeyValueTuple(mlirModule, llvmModule,
+                                     mdTuple->getOperand(summayIdx++),
+                                     "MaxFunctionCount", maxFunctionCount))
+    return nullptr;
+  if (!convertInt64FromKeyValueTuple(mlirModule, llvmModule,
+                                     mdTuple->getOperand(summayIdx++),
+                                     "NumCounts", numCounts))
+    return nullptr;
+  if (!convertInt64FromKeyValueTuple(mlirModule, llvmModule,
+                                     mdTuple->getOperand(summayIdx++),
+                                     "NumFunctions", numFunctions))
+    return nullptr;
+
+  // Handle optional keys.
+  std::optional<uint64_t> isPartialProfile;
+  if (getOptIntValue(mdTuple->getOperand(summayIdx), "IsPartialProfile",
+                     isPartialProfile)
+          .failed())
+    return nullptr;
+  if (isPartialProfile.has_value())
+    summayIdx++;
+
+  FloatAttr partialProfileRatio;
+  if (getOptDoubleValue(mdTuple->getOperand(summayIdx), "PartialProfileRatio",
+                        partialProfileRatio)
+          .failed())
+    return nullptr;
+  if (partialProfileRatio)
+    summayIdx++;
+
+  // Handle detailed summary.
+  SmallVector<ModuleFlagProfileSummaryDetailedAttr> detailed;
+  if (!convertProfileSummaryDetailed(mlirModule, llvmModule,
+                                     mdTuple->getOperand(summayIdx), detailed))
+    return nullptr;
+
+  // Build the final profile summary attribute.
+  return ModuleFlagProfileSummaryAttr::get(
+      mlirModule->getContext(), *format, totalCount, maxCount, maxInternalCount,
+      maxFunctionCount, numCounts, numFunctions, isPartialProfile,
+      partialProfileRatio ? partialProfileRatio : nullptr, detailed);
```

Comment:
```suggestion
      partialProfileRatio, detailed);
```
nit: I think if it is nullptr it is nullptr ?